src/main/java/nurseybook/logic/parser/ViewScheduleCommandParser.java

```diff
@@ -20,16 +23,32 @@
      * @throws ParseException if the user input does not conform the expected format
      */
     public ViewScheduleCommand parse(String args) throws ParseException {
-        String trimmedArgs = args.trim();
+        String trimmedKeyDate = args.trim();
 
-        if (trimmedArgs.isEmpty()) {
+        if (trimmedKeyDate.isEmpty()) {
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, ViewScheduleCommand.MESSAGE_USAGE));
         }
 
-        String stringKeyDate = trimmedArgs.split(" ")[0];
-        LocalDate keyDate = ParserUtil.parseDate(stringKeyDate);
+        try {
+            LocalDate keyDate = ParserUtil.parseDate(trimmedKeyDate);
+            isDateWithinBounds(keyDate);
+            return new ViewScheduleCommand(new DateTimeContainsDatePredicate(keyDate), keyDate);
+        } catch (ParseException pe) {
+            if (pe.getMessage().equals(MESSAGE_VIEWSCHEDULE_DAYS_SUPPORTED)) {
+                throw pe;
+            } else {
+                throw new ParseException(
+                        String.format(MESSAGE_INVALID_COMMAND_FORMAT, ViewScheduleCommand.MESSAGE_USAGE), pe);
```

Comment:
Hmm I'm not sure whether it would be better to throw the invalid command format message or invalid date message here. The task methods that expect a date parameter throws the invalid date message. I think it might be better for the error message to be about invalid dates? The invalid date message also shows the correct date format that the user should enter. Then we can just let the method throw the `ParseException` and remove the try-catch block too.

Reply:
Good suggestion! I will update accordingly.